copy/app.js

```diff
@@ -0,0 +1,141 @@
+        <li><i>GET /secretgist/ID</i>: retrieve and decrypt a given secret gist
+        <li><i>POST /create { name, content }</i>: create a private gist for the authorized user with given name/content</li>
+        <li><i>POST /createsecret { name, content }</i>: create a private and encrypted gist for the authorized user with given name/content</li>
+      </ul>
+      <h3>Create an *unencrypted* gist</h3>
+      <form action="/create" method="post">
+        Name: <input type="text" name="name"><br>
+        Content:<br><textarea name="content" cols="80" rows="10"></textarea><br>
+        <input type="submit" value="Submit">
+      </form>
+      <h3>Create an *encrypted* gist</h3>
+      <form action="/createsecret" method="post">
+        Name: <input type="text" name="name"><br>
+        Content:<br><textarea name="content" cols="80" rows="10"></textarea><br>
+        <input type="submit" value="Submit">
+      </form>
+    </body>
+  </html>
+`);
+});
+  // TODO Retrieve a list of all gists for the currently authed user
+server.get('/gists', (req, res) => {
+  github.users.getForUser({ username: `${username}` })
+  .then((response) => {
+    res.json(response.data);
+  })
+  .catch((err) => {
+    res.json({ meassge: 'no gists found', err });
+  });
+});
+
+server.get('/key', (req, res) => {
+  // TODO Return the secret key used for encryption of secret gists
+  res.send(nacl.util.encodeBase64(genKey));
+});
+
+server.get('/secretgist/:id', (req, res) => {
+  // TODO Retrieve and decrypt the secret gist corresponding to the given ID
+  const id = req.params.id;
+  github.gists.get({ id })
+    .then((response) => {
+      const gist = response.data;
+      const filename = Object.keys(gist.files)[0];
+      const blob = gist.files[filename].content;
+      const nonce = nacl.util.decodeBase64(blob.slice(0, 32));
+      const ciphertext = nacl.util.decodeBase64(blob.slice(32, blob.length));
+      const plaintext = nacl.secretbox.open(ciphertext, nonce, genKey);
+      res.send(nacl.util.encodeUTF8(plaintext));
+    });
+});
+
+server.post('/create', urlencodedParser, (req, res) => {
+  // TODO Create a private gist with name and content given in post request
+  const { name, content } = req.body;
+  const files = { name: content };
+  github.gists.create({ files, public: false })
+    .then((response) => {
+      return res.json(response.data);
+    })
+      .catch((err) => {
+        res.json(err);
+      });
+});
+
+server.post('/createsecret', (req, res) => {
+  // TODO Create a private and encrypted gist with given name/content
+  // NOTE - we're only encrypting the content, not the filename
+  // To save, we need to keep both encrypted content and nonce
+  const { name, content } = req.body;
+  const nonce = nacl.randomBytes(24);
+  const ciphertext = nacl.secretbox(nacl.util.decodeUTF8(content), nonce, genKey);
+  const blob = nacl.util.encodeBase64(nonce) + nacl.encodeBase64(ciphertext);
```

Comment:
`nacl.encodeBase64(ciphertext)` should actually be `nacl.util.encodeBase64(ciphertext)`.